src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.NativeAot.Windows.cs

```diff
@@ -13,19 +13,22 @@ private static IntPtr LoadLibraryHelper(string libraryName, int flags, ref LoadL
         {
             IntPtr hmod;
 
+            // Disable the OS dialogs when failing to load. This matches CoreCLR.
+            uint prev;
+            bool set = Interop.Kernel32.SetThreadErrorMode(Interop.Kernel32.SEM_FAILCRITICALERRORS | Interop.Kernel32.SEM_NOOPENFILEERRORBOX, out prev);
```

Comment:
Will update CoreCLR to use `SetThreadErrorMode` instead of `SetErrorMode` in a future PR.

Reply:
See https://github.com/dotnet/runtime/pull/90122